test-network-function/networking/suite_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) 2020-2022 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package networking
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/test-network-function/test-network-function/pkg/tnf/interactive"
+)
+
+func TestParseVariables(t *testing.T) {
+	// expected inputs
+	testCases := []struct {
+		// inputs
+		inputRes      string
+		declaredPorts map[key]string
+
+		// expected outputs here
+		expectedDeclaredPorts map[key]string
+		exceptedRes           string
+	}{
+		{
+			inputRes:              `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"TCP\"\n  }\n]`,
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{},
+			exceptedRes:           `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"TCP\"\n  }\n]`,
+		},
+	}
```

Comment:
We should have some more testcases here like what happens if the `inputRes` is empty string.